src/traces/table/plot.js

```diff
@@ -0,0 +1,892 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var c = require('./constants');
+var d3 = require('d3');
+var gup = require('../../lib/gup');
+var Drawing = require('../../components/drawing');
+var extendFlat = require('../../lib/extend').extendFlat;
+var svgUtil = require('../../lib/svg_text_utils');
+var raiseToTop = require('../../lib').raiseToTop;
+var cancelEeaseColumn = require('../../lib').cancelTransition;
+
+module.exports = function plot(gd, calcdata) {
+
+    if(c.clipView) {
+        gd._fullLayout._paper.attr('height', 2000);
+    }
+
+    var table = gd._fullLayout._paper.selectAll('.table')
+        .data(calcdata.map(gup.unwrap), gup.keyFun);
+
+    table.exit().remove();
+
+    table.enter()
+        .append('g')
+        .classed('table', true)
+        .attr('overflow', 'visible')
+        .style('box-sizing', 'content-box')
+        .style('position', 'absolute')
+        .style('left', 0)
+        .style('overflow', 'visible')
+        .style('shape-rendering', 'crispEdges')
+        .style('pointer-events', 'all'); // todo restore 'none'
+
+    table
+        .attr('width', function(d) {return d.width + d.size.l + d.size.r;})
+        .attr('height', function(d) {return d.height + d.size.t + d.size.b;})
+        .attr('transform', function(d) {
+            return 'translate(' + d.translateX + ',' + d.translateY + ')';
+        });
+
+    var tableControlView = table.selectAll('.tableControlView')
+        .data(gup.repeat, gup.keyFun);
+
+    tableControlView.enter()
+        .append('g')
+        .classed('tableControlView', true)
```

Comment:
`table-control-view` - use dash-case for class names please.

Reply:
... and those class names would be best defined in `table/constants.js`

Reply:
I'll do the classname conversion to dashes, serves code unity! Seen it both w/ dashes and even all-lowercase w/o separators in `plotly.js`. Btw. if there's technical merit pro/cons I'd be interested to know too.

Glad to refactor classnames into `constants` and also, the selection `'.classname'`s can be glued together from that. I'd probably leave the this 2nd thing to the end so higher prio changes can be done 1st (worst case, minor, isolated follow-up PR w/ extraction, also including the same dashing+extraction for `parcoords`).

Reply:
Included it in https://github.com/plotly/plotly.js/issues/2056